### PR TITLE
MF-797: Add swr to medications widget

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR:

1. Adds SWR data fetching hooks to the Medications widget.
2. Refactors the existing components to reuse these hooks.
3. Refactors existing tests to work within the SWR paradigm.

## Issue

https://issues.openmrs.org/browse/MF-797

### CODE REVIEW

**Comment by HerbertYiga** [comment]
cc @FlorianRappl @brandones

**Review by brandones — APPROVED** [review]

**Comment by brandones** [comment]
Sorry it took so long that we've got conflicts, @denniskigen . Feel free to merge after resolving.

File: packages/esm-patient-medications-app/src/api/api.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export async function fetchPatientOrders(
   abortController?: AbortController,
 ) {
   const { data } = await openmrsFetch<PatientMedicationFetchResponse>(
```
Comment (brandones) [inline packages/esm-patient-medications-app/src/api/api.ts:53]: Oh wait, can't we delete this whole function `fetchPatientOrders` function now?

Comment (denniskigen) [reply packages/esm-patient-medications-app/src/api/api.ts:53]: Correct! I've moved the hook `usePatientOrders` hook into the `api.ts` file as well for consistency.

**Comment by denniskigen** [comment]
Thanks, @brandones!

**Pull request merged**